packages/flutter/lib/src/widgets/slotted_render_object_widget.dart

```diff
@@ -231,14 +240,47 @@ class SlottedRenderObjectElement<S> extends RenderObjectElement {
     }(), '${widget.runtimeType}.slots must not change.');
     assert(slottedMultiChildRenderObjectWidgetMixin.slots.toSet().length == slottedMultiChildRenderObjectWidgetMixin.slots.length, 'slots must be unique');
 
+    final Map<GlobalKey, S> oldGlobalKeyedSlots = _globalKeyToSlot;
+    _globalKeyToSlot = <GlobalKey, S>{};
+
+    // New slots for children that are moved to a different slot by global key
+    // reparenting. It shouldn't be updated in place because two children could
+    // swap slots.
+    final Map<S, Element> slotToKeyedChild = <S, Element>{};
+
+    // Process children that need to be moved to new slots because of global
+    // key reparenting first.
     for (final S slot in slottedMultiChildRenderObjectWidgetMixin.slots) {
-      _updateChild(slottedMultiChildRenderObjectWidgetMixin.childForSlot(slot), slot);
+      final Widget? widget = slottedMultiChildRenderObjectWidgetMixin.childForSlot(slot);
+      final Key? newWidgetKey = widget?.key;
+      if (widget != null && newWidgetKey is GlobalKey) {
```

Comment:
Regarding comments from the original pull request:

> This now has me wondering: Don't we need to do this for all keys, not just globalkeys? If a keyed widget changes slots, it should be matched with the element from the previous slot it was in?

The documentation seems to suggest that should be the case:
```
/// A key that is not a [GlobalKey].
///
/// Keys must be unique amongst the [Element]s with the same parent. By
/// contrast, [GlobalKey]s must be unique across the entire app.
```

But I feel it to an extent overlaps the idea of the children being "slotted"

Reply:
I was under the impression that to be able to use a local key the children have to be somewhat homogeneous, like in a list view or a column, but that doesn't seem to be true according to the documentation.

Reply:
Replaced GlobalKey with Key.